### ISSUE TITLE
DriveLetter_Unix currently failing

### ISSUE THREAD

**Issue by mmitche** [dotnet/corefx#2459]
System.IO.FileSystem.Tests.Directory_CreateDirectory.DriveLetter_Unix [FAIL]
09:58:09       System.IO.DirectoryNotFoundException : Could not find a part of the path '/mnt/j/workspace/dotnet_corefx_linux_debug_tst_prtest/bin/tests/Windows_NT.AnyCPU.Debug/System.IO.FileSystem.Tests/dnxcore50/C:'.
09:58:09       Stack Trace:
09:58:10             at System.IO.UnixFileSystem.RemoveDirectoryInternal(String fullPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
09:58:10             at System.IO.UnixFileSystem.RemoveDirectory(String fullPath, Boolean recursive)
09:58:10             at System.IO.Directory.Delete(String path)
09:58:10             at System.IO.FileSystem.Tests.Directory_CreateDirectory.DriveLetter_Unix()
09:58:10 Finished:    System.IO.FileSystem.Tests

**Comment by ianhays** [comment]
I don't think that this is actually a failing test; it passes fine on my computer with 100 iterations. Something odd must have happened to the current directory of the build system to make the test fail in this particular way.

I'm going to reenable it in my next PR and it should pass through Jenkins without issue.

**Comment by mmitche** [comment]
@ianhays Thanks

**Comment by stephentoub** [comment]
This is failing again.  This is the second time I've seen it fail now in the last two days:
```
20:11:37    System.IO.Tests.Directory_CreateDirectory.DriveLetter_Unix [FAIL]
20:11:37       System.IO.DirectoryNotFoundException : Could not find a part of the path '/mnt/resource/j/workspace/dotnet_corefx/debian8.2_release_tst/bin/tests/Linux.AnyCPU.Release/System.IO.FileSystem.Tests/dnxcore50/C:'.
20:11:37       Stack Trace:
20:11:38             at System.IO.UnixFileSystem.RemoveDirectoryInternal(DirectoryInfo directory, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
20:11:38             at System.IO.UnixFileSystem.RemoveDirectory(String fullPath, Boolean recursive)
```
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/debian8.2_release_tst/775/console

**Comment by ianhays** [comment]
This is resolved by #7308. I ran into this in CI on that PR.
